transport/socks5/socks5.go

```diff
@@ -119,3 +127,47 @@ func appendSOCKS5Address(b []byte, address string) ([]byte, error) {
 	b = binary.BigEndian.AppendUint16(b, uint16(portNum))
 	return b, nil
 }
+
+func readAddress(reader io.Reader) (string, int, error) {
```

Comment:
Please add tests for this function.

Reply:
@fortuna readAddress changed to returned address struct and tests added.